data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/parser/ModelConvertingLogstashVisitor.java

```diff
@@ -96,11 +96,22 @@ else if (attributeContext.value().getChild(0) instanceof LogstashParser.PluginCo
         }
         else if (attributeContext.value().NUMBER() != null && attributeContext.value().getText().equals(attributeContext.value().NUMBER().toString())) {
             logstashValueType = LogstashValueType.NUMBER;
-            value = Double.valueOf(attributeContext.value().getText());
+            try {
+                value = Integer.parseInt(attributeContext.value().getText());
+            } catch (NumberFormatException e) {
+                try {
+                    value = Double.parseDouble(attributeContext.value().getText());
+                } catch (NumberFormatException exception) {
+
+                }
+            }
         }
         else if (attributeContext.value().BAREWORD() != null && attributeContext.value().getText().equals(attributeContext.value().BAREWORD().toString())) {
             logstashValueType = LogstashValueType.BAREWORD;
             value = attributeContext.value().getText();
+            if (value.toString().equals("true") || value.toString().equals("false")) {
```

Comment:
Should we do a case-insensitive equality here? If Logstash does that, I'd like to as well. But, if the values must be all lowercase, then what you have is perfect.

Reply:
Logstash only supports all lowercase `true` or `false`

Reply:
Thanks for confirming that!